linkcheck/configuration/__init__.py

```diff
@@ -428,43 +428,45 @@ def get_user_config():
     return userconf
 
 
-def get_gconf_http_proxy():
-    """Return host:port for GConf HTTP proxy if found, else None."""
+def get_dconf_http_proxy():
+    """Return host:port for dconf HTTP proxy if found, else None."""
     try:
-        import gconf
+        import gi
+        from gi.repository import Gio
     except ImportError:
         return None
     try:
-        client = gconf.client_get_default()
-        if client.get_bool("/system/http_proxy/use_http_proxy"):
-            host = client.get_string("/system/http_proxy/host")
-            port = client.get_int("/system/http_proxy/port")
+        settings = Gio.Settings.new("org.gnome.system.proxy.http")
+        if settings.get_boolean("enabled"):
+            host = settings.get_string("host")
+            port = settings.get_int("port")
             if host:
                 if not port:
                     port = 8080
                 return "%s:%d" % (host, port)
     except Exception as msg:
-        log.debug(LOG_CHECK, "error getting HTTP proxy from gconf: %s", msg)
+        log.debug(LOG_CHECK, "error getting HTTP proxy from dconf: %s", msg)
         pass
     return None
 
 
-def get_gconf_ftp_proxy():
-    """Return host:port for GConf FTP proxy if found, else None."""
+def get_dconf_ftp_proxy():
+    """Return host:port for dconf FTP proxy if found, else None."""
     try:
-        import gconf
+        import gi
+        from gi.repository import Gio
     except ImportError:
         return None
     try:
-        client = gconf.client_get_default()
-        host = client.get_string("/system/proxy/ftp_host")
-        port = client.get_int("/system/proxy/ftp_port")
+        settings = Gio.Settings.new("org.gnome.system.proxy.ftp")
+        host = settings.get_string("host")
+        port = settings.get_int("port")
```

Comment:
This just begs for code reuse, although, hmm, the http proxy has an enabled flag (and a bunch of authentication options what we ignore?), but the ftp (and https) one doesn't.

Reply:
Done.